### PR TITLE
fix: handle `max_heap_size` = 0

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://github.com/emqx/emqx/issues/13417

Fixes https://emqx.atlassian.net/browse/EMQX-12659

Release version: v/e5.7.2

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_utils/src/emqx_utils.erl [hunk 1]
```diff
@@ -289,8 +289,10 @@ do_check_oom([{Val, Max, Reason} | Rest]) ->
     end.
 
 tune_heap_size(#{enable := false}) ->
-    ok;
+    ignore;
 %% If the max_heap_size is set to zero, the limit is disabled.
+tune_heap_size(#{max_heap_size := 0}) ->
+    ignore;
```
Comment (keynslug) [inline apps/emqx_utils/src/emqx_utils.erl:295]: There's currently no mention that setting 0 is equivalent to disabling. I guess this needs to be mentioned in `emqx_schema.hocon`?

**Review by keynslug — APPROVED** [review]

File: apps/emqx_utils/src/emqx_utils.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_utils/src/emqx_utils.erl:295]: True, there's only this comment and the [originating issue](https://github.com/emqx/emqx/issues/13417).

Comment (thalesmg) [reply apps/emqx_utils/src/emqx_utils.erl:295]: :checkered_flag:

**Review by keynslug — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2904)

**Pull request merged**